index.js

```diff
@@ -103,6 +112,45 @@ function callCerberus (context, type, keyPath, data, cb) {
   }
 }
 
+function fileCerberus (context, type, filePath, data, cb) {
+  context.log(`Starting ${type} file request for ${filePath}`)
+  let action = co(function * () {
+    let token = yield getToken(context)
+    if (!token) throw new Error('unable to retrieve token')
+
+    let url = urlJoin(context.hostUrl, cerberusVersion, type === 'LIST' ? 'secure-files' : 'secure-file', filePath)
+    context.log('token retrieved', token, filePath, url)
+
+    let form
+    if (data) {
+      form = new FormData()
+      form.append('file-content', data, {filename: filePath.match(/([^\/]*)\/*$/)[1]})
+    }
+    let fileResponse = yield request({
+      method: type === 'LIST' ? 'GET' : type,
+      url: url,
+      headers: Object.assign({}, globalHeaders, {'X-Vault-Token': token}, type === 'POST' ? form.getHeaders() : undefined),
+      body: form,
+      json: type === 'LIST' || type === 'DELETE'
+    })
+
+    let fileResult = fileResponse.data
+    if (fileResult && fileResult.errors && fileResult.errors.length > 0) throw new Error(formatCerberusError(fileResult.errors))
+
+    context.log('file retrieved', fileResponse.statusCode.toString(), fileResult)
+    if (fileResponse.statusCode && fileResponse.statusCode.toString()[0] !== '2') throw new Error('File Request error, Status: ' + fileResponse.statusCode)
```

Comment:
How does this not fail a linter?

Please use curly braces for `if statements` and google all the very famous bugs that happen from people not using curly braces on if statements.

https://blog.codecentric.de/en/2014/02/curly-braces/

- [ ] If statements should have curly braces.

Reply:
Standard allows for un-braced `if` statements if the block is on the same line as the `if` statement. See [the multi option on the curly rule](https://eslint.org/docs/rules/curly)